src/main/java/monday/storage/Storage.java

```diff
@@ -0,0 +1,74 @@
+package monday.storage;
+
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+import monday.task.Task;
+
+
+/**
+ * Class for handling the storage of tasks in a file.
+ */
+public class Storage {
+    //CHECKSTYLE.OFF: AbbreviationAsWordInName
+    //CHECKSTYLE.OFF: MemberName
+    private final String FILEPATH;
+    //CHECKSTYLE.ON: AbbreviationAsWordInName
+    //CHECKSTYLE.ON: MemberName
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param fileName the name of the file to store the tasks
+     */
+    public Storage(String fileName) {
+        this.FILEPATH = fileName;
+        File file = new File(fileName);
+        if (!file.getParentFile().exists()) {
+            file.getParentFile().mkdirs();
+        }
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                System.out.println("Error creating file: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Saves the given list of tasks to the file.
+     *
+     * @param tasks the list of tasks to save
+     * @throws IOException if an I/O error occurs while saving the tasks
+     */
+    public void save(ArrayList<Task> tasks) throws IOException {
+        try (ObjectOutputStream output = new ObjectOutputStream(new FileOutputStream(FILEPATH))) {
+            output.writeObject(tasks);
+        }
```

Comment:
```suggestion
        output.writeObject(tasks);
```
This might be a bit redundant
(p.s. trying the suggestion feature; feel free to ignore / close / reject)

Reply:
Good catch, did not realise the redundant try block. Thank you!